In [2]:
import requests

In [3]:
res1 = requests.get("https://api.frankfurter.dev/v1/latest?from=USD").json()
base = res1.get("base")
date = res1.get("date")
total_currencies = len(res1.get("rates", {}))

print(f"Task 1 Results:")
print(f"Base: {base} | Date: {date} | Currencies Returned: {total_currencies}\n")



Task 1 Results:
Base: USD | Date: 2026-09-17 | Currencies Returned: 29



In [4]:
res2 = requests.get("https://api.frankfurter.dev/v1/currencies").json()
print("Task 2 Results:")
print(f"Total Supported Currencies: {len(res2)}")
print(f"Is NGN supported? {'NGN' in res2}")
print("Supported Currencies:")
for code, name in res2.items():
    print(f" {code}: {name}")
print("\n")


Task 2 Results:
Total Supported Currencies: 30
Is NGN supported? False
Supported Currencies:
 AUD: Australian Dollar
 BRL: Brazilian Real
 CAD: Canadian Dollar
 CHF: Swiss Franc
 CNY: Chinese Renminbi Yuan
 CZK: Czech Koruna
 DKK: Danish Krone
 EUR: Euro
 GBP: British Pound
 HKD: Hong Kong Dollar
 HUF: Hungarian Forint
 IDR: Indonesian Rupiah
 ILS: Israeli New Shekel
 INR: Indian Rupee
 ISK: Icelandic Króna
 JPY: Japanese Yen
 KRW: South Korean Won
 MXN: Mexican Peso
 MYR: Malaysian Ringgit
 NOK: Norwegian Krone
 NZD: New Zealand Dollar
 PHP: Philippine Peso
 PLN: Polish Złoty
 RON: Romanian Leu
 SEK: Swedish Krona
 SGD: Singapore Dollar
 THB: Thai Baht
 TRY: Turkish Lira
 USD: United States Dollar
 ZAR: South African Rand




In [5]:
def get_rates(base_currency):
    try:
        url = f"https://api.frankfurter.dev/v1/latest?from={base_currency}"
        response = requests.get(url, timeout=10)
        
        if response.status_code == 200:
            return response.json()
        else:
            # Handle API error response gracefully
            error_data = response.json() if response.headers.get("content-type") == "application/json" else {}
            msg = error_data.get("message", "Not found or invalid currency")
            return f"Failed (Status {response.status_code}): {msg}"
    except Exception as e:
        return f"Request failed: {str(e)}"

currencies_to_test = ["USD", "EUR", "GBP", "JPY", "CHF", "NGN", "ZWL", "XYZ", "ng"]

print("Task 3 Results:")
for code in currencies_to_test:
    print(f"{code}: {get_rates(code)}")
print("\n")



Task 3 Results:
USD: {'amount': 1.0, 'base': 'USD', 'date': '2026-09-17', 'rates': {'AUD': 1.4057, 'BRL': 5.1307, 'CAD': 1.3995, 'CHF': 0.82449, 'CNY': 6.7075, 'CZK': 21.172, 'DKK': 6.511, 'EUR': 0.871, 'GBP': 0.74758, 'HKD': 7.8452, 'HUF': 316.16, 'IDR': 17727, 'ILS': 3.0333, 'INR': 95.94, 'ISK': 121.77, 'JPY': 155.69, 'KRW': 1382.55, 'MXN': 17.1881, 'MYR': 4.099, 'NOK': 9.4286, 'NZD': 1.7414, 'PHP': 62.718, 'PLN': 3.7958, 'RON': 4.5836, 'SEK': 9.8175, 'SGD': 1.2753, 'THB': 33.305, 'TRY': 48.675, 'ZAR': 16.2525}}
EUR: {'amount': 1.0, 'base': 'EUR', 'date': '2026-09-17', 'rates': {'AUD': 1.6139, 'BRL': 5.8905, 'CAD': 1.6068, 'CHF': 0.9466, 'CNY': 7.7009, 'CZK': 24.308, 'DKK': 7.4753, 'GBP': 0.8583, 'HKD': 9.0071, 'HUF': 362.98, 'IDR': 20352.48, 'ILS': 3.4825, 'INR': 110.1475, 'ISK': 139.8, 'JPY': 178.75, 'KRW': 1587.31, 'MXN': 19.7337, 'MYR': 4.7061, 'NOK': 10.825, 'NZD': 1.9993, 'PHP': 72.006, 'PLN': 4.358, 'RON': 5.2624, 'SEK': 11.2715, 'SGD': 1.4642, 'THB': 38.237, 'TRY': 55.8841, '

In [6]:
users_data = requests.get("https://jsonplaceholder.typicode.com/users").json()

users_dict = {
    user["id"]: {
        "name": user["name"],
        "email": user["email"],
        "city": user["address"]["city"]
    }
    for user in users_data
}

print("Task 4 Results:")
print(users_dict)
print("\n")

Task 4 Results:
{1: {'name': 'Leanne Graham', 'email': 'Sincere@april.biz', 'city': 'Gwenborough'}, 2: {'name': 'Ervin Howell', 'email': 'Shanna@melissa.tv', 'city': 'Wisokyburgh'}, 3: {'name': 'Clementine Bauch', 'email': 'Nathan@yesenia.net', 'city': 'McKenziehaven'}, 4: {'name': 'Patricia Lebsack', 'email': 'Julianne.OConner@kory.org', 'city': 'South Elvis'}, 5: {'name': 'Chelsey Dietrich', 'email': 'Lucio_Hettinger@annie.ca', 'city': 'Roscoeview'}, 6: {'name': 'Mrs. Dennis Schulist', 'email': 'Karley_Dach@jasper.info', 'city': 'South Christy'}, 7: {'name': 'Kurtis Weissnat', 'email': 'Telly.Hoeger@billy.biz', 'city': 'Howemouth'}, 8: {'name': 'Nicholas Runolfsdottir V', 'email': 'Sherwood@rosamond.me', 'city': 'Aliyaview'}, 9: {'name': 'Glenna Reichert', 'email': 'Chaim_McDermott@dana.io', 'city': 'Bartholomebury'}, 10: {'name': 'Clementina DuBuque', 'email': 'Rey.Padberg@karina.biz', 'city': 'Lebsackbury'}}




In [7]:
post_payload = {"title": "New Post", "body": "Sample content", "userId": 1}
post_res = requests.post("https://jsonplaceholder.typicode.com/posts", json=post_payload)

status_code = post_res.status_code
created_id = post_res.json().get("id")

get_res = requests.get(f"https://jsonplaceholder.typicode.com/posts/{created_id}")

print("Task 5 Results:")
print(f"POST Status Code: {status_code} | Generated ID: {created_id}")
print(f"GET Status Code for ID {created_id}: {get_res.status_code}")


Task 5 Results:
POST Status Code: 201 | Generated ID: 101
GET Status Code for ID 101: 404


For number 5: Fetching post id 101 returns HTTP 404 Not Found response. This happens because Jsonplaceholder is a mock rest api that fakes creation responses without persisting any new resources to a server database